course/src/main/java/in/testpress/course/pagination/RunningContentRemoteMediator.kt

```diff
@@ -0,0 +1,115 @@
+package `in`.testpress.course.pagination
+
+import `in`.testpress.database.dao.RunningContentDao
+import `in`.testpress.database.dao.RunningContentRemoteKeysDao
+import `in`.testpress.database.entities.RunningContentEntity
+import `in`.testpress.database.entities.RunningContentRemoteKeys
+import androidx.paging.*
+import androidx.room.withTransaction
+import `in`.testpress.course.network.CourseNetwork
+import `in`.testpress.database.TestpressDatabase
+import `in`.testpress.v2_4.models.ApiResponse
+import retrofit2.HttpException
+import java.io.IOException
+
+private const val DEFAULT_PAGE_INDEX = 1
+
+@OptIn(ExperimentalPagingApi::class)
+class RunningContentRemoteMediator(
+    val courseNetwork: CourseNetwork,
+    val database: TestpressDatabase,
+    val courseId: Long
+) : RemoteMediator<Int, RunningContentEntity>() {
+
+    private val runningContentDao: RunningContentDao = database.runningContentDao()
+    private val runningContentRemoteKeysDao: RunningContentRemoteKeysDao = database.runningContentRemoteKeysDao()
+    private var endOfPaginationReached = true
+    private var pageNumber = 1
+
+    override suspend fun initialize(): InitializeAction {
+        return InitializeAction.LAUNCH_INITIAL_REFRESH
+    }
+
+    override suspend fun load(
+        loadType: LoadType,
+        state: PagingState<Int, RunningContentEntity>
+    ): MediatorResult {
+
+        try {
+            pageNumber = when (loadType) {
+                LoadType.REFRESH -> DEFAULT_PAGE_INDEX
+                LoadType.PREPEND -> return MediatorResult.Success(endOfPaginationReached = true)
+                LoadType.APPEND -> {
+                    getNextPageNumber(state)
+                        ?: return MediatorResult.Success(endOfPaginationReached = true)
+                }
+            }
```

Comment:
This can be extracted to separate method. Let it return -1 for prepend or no next page